pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -377,7 +382,157 @@ func ValidateKubernetesVersionSkew(new, old *Cluster) field.ErrorList {
+func ValidateWorkerKubernetesVersionSkew(new, old *Cluster) field.ErrorList {
+	var allErrs field.ErrorList
+
+	topKubeVersion := new.Spec.KubernetesVersion
+	oldTopKubeVersion := old.Spec.KubernetesVersion
+
+	workerNodeGroupMap := make(map[string]*WorkerNodeGroupConfiguration)
+	for i := range old.Spec.WorkerNodeGroupConfigurations {
+		workerNodeGroupMap[old.Spec.WorkerNodeGroupConfigurations[i].Name] = &old.Spec.WorkerNodeGroupConfigurations[i]
+	}
+	for _, nodeGroupNewSpec := range new.Spec.WorkerNodeGroupConfigurations {
+		if workerNodeGrpOldSpec, ok := workerNodeGroupMap[nodeGroupNewSpec.Name]; ok {
+			oldVersion := workerNodeGrpOldSpec.KubernetesVersion
+			newVersion := nodeGroupNewSpec.KubernetesVersion
+
+			allErrs = performWorkerKubernetesValidations(oldVersion, newVersion, oldTopKubeVersion, topKubeVersion)
+		}
+	}
+
+	return allErrs
+}
+
+func performWorkerKubernetesValidations(oldVersion, newVersion *KubernetesVersion, oldTopKubeVersion, topKubeVersion KubernetesVersion) field.ErrorList {
+	var allErrs field.ErrorList
+	path := field.NewPath("spec").Child("WorkerNodeConfiguration.kubernetesVersion")
+
+	if oldVersion != nil && newVersion == nil {
+		allErrs = append(
+			allErrs,
+			validateRemoveWorkerKubernetesVersion(topKubeVersion, oldTopKubeVersion, newVersion, oldVersion)...,
+		)
+	}
+
+	if oldVersion != nil && newVersion != nil {
+		allErrs = append(
+			allErrs,
+			validateKubeVersionSkew(*newVersion, *oldVersion, path)...,
+		)
+	}
+
+	if oldVersion == nil && newVersion != nil {
+		allErrs = append(
+			allErrs,
+			validateKubeVersionSkew(*newVersion, oldTopKubeVersion, path)...,
+		)
+	}
+
+	if newVersion != nil {
+		allErrs = append(
+			allErrs,
+			validateCPWorkerKubeSkew(topKubeVersion, *newVersion)...,
+		)
+	}
+
+	return allErrs
+}
+
+func validateRemoveWorkerKubernetesVersion(newCPVersion, oldCPVersion KubernetesVersion, newWorkerVersion, oldWorkerVersion *KubernetesVersion) field.ErrorList {
+	var allErrs field.ErrorList
+	path := field.NewPath("spec").Child("WorkerNodeConfiguration.kubernetesVersion")
+
+	parsedWorkerVersion, err := version.ParseGeneric(string(*oldWorkerVersion))
+	if err != nil {
+		allErrs = append(
+			allErrs,
+			field.Invalid(path, newWorkerVersion, fmt.Sprintf("could not parse version: %v, %v", oldWorkerVersion, err)))
+		return allErrs
+	}
+
+	parsedOldVersion, err := version.ParseGeneric(string(oldCPVersion))
+	if err != nil {
+		allErrs = append(
+			allErrs,
+			field.Invalid(path, newWorkerVersion, fmt.Sprintf("could not parse version: %v, %v", oldCPVersion, err)))
+		return allErrs
+	}
+
+	parsedNewVersion, err := version.ParseGeneric(string(newCPVersion))
+	if err != nil {
+		allErrs = append(
+			allErrs,
+			field.Invalid(path, newWorkerVersion, fmt.Sprintf("could not parse version: %v, %v", newCPVersion, err)))
+		return allErrs
+	}
+
+	if parsedOldVersion.LessThan(parsedNewVersion) {
```

Comment:
There is no Equal function and parsedNewVersion can only be greater than or equal to parsedOldVersion (these are both cluster level versions). The validations for the cluster level kubernetes version happens before worker nodes so if there was an issue at the cluster level version, then it would be caught there.

Reply:
Is this not the [Equal](https://github.com/tatlat/eks-anywhere/blob/49b4ae59fdd898f40d9c29e27be1908014b2570b/pkg/semver/semver.go#L67) function?

But yeah, it does not really matter, I was mainly wondering why `LessThan` over simply using the `Equal` comparison.